core/shared/src/main/scala/com/softwaremill/sttp/SttpBackendOptions.scala

```diff
@@ -21,8 +23,24 @@ case class SttpBackendOptions(
 }
 
 object SttpBackendOptions {
-  case class Proxy(host: String, port: Int, proxyType: ProxyType) {
-    def asJava = new java.net.Proxy(proxyType.asJava, inetSocketAddress)
+  case class Proxy(host: String, port: Int, proxyType: ProxyType, nonProxyHosts: List[String] = Nil) {
+    def asJavaProxySelector: net.ProxySelector = new net.ProxySelector {
+      override def select(uri: URI): util.List[net.Proxy] = {
+        val proxyList = new util.ArrayList[net.Proxy](1)
+        val uriHost = uri.getHost
+        if (uriHost.startsWith("127.0.0.1") || nonProxyHosts.forall(uriHost.startsWith)) {
```

Comment:
Shouldn't this be an `exists`? If so, maybe this can be factored out into a method on `Proxy` (there's another similar piece of code in `HttpURLConnectioBackend`)

Reply:
I'm still unsure if I should be using startsWith or plain equals. I guess plain equals should work.